internal/backend/runtime/omni/controllers/omni/internal/machineset/operations.go

```diff
@@ -205,16 +206,19 @@ func (d *Destroy) Apply(ctx context.Context, r controller.ReaderWriter, logger *
 	return nil
 }
 
-func setPatches(clusterMachineConfigPatches *omni.ClusterMachineConfigPatches, patches []*omni.ConfigPatch) {
+func setPatches(clusterMachineConfigPatches *omni.ClusterMachineConfigPatches, patches []*omni.ConfigPatch, compressConfigData bool) error {
 	patchesRaw := make([]string, 0, len(patches))
 
 	for _, patch := range patches {
-		data := patch.TypedSpec().Value.Data
+		data, err := patch.TypedSpec().Value.GetUncompressedData()
```

Comment:
be :cool: and don't recompress patches if they are already compressed

Reply:
done